pkg/controller/nodesetup/controller.go

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2023 ScyllaDB.
+
+package nodesetup
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	scyllav1alpha1client "github.com/scylladb/scylla-operator/pkg/client/scylla/clientset/versioned/typed/scylla/v1alpha1"
+	scyllav1alpha1informers "github.com/scylladb/scylla-operator/pkg/client/scylla/informers/externalversions/scylla/v1alpha1"
+	scyllav1alpha1listers "github.com/scylladb/scylla-operator/pkg/client/scylla/listers/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
+	"github.com/scylladb/scylla-operator/pkg/kubeinterfaces"
+	"github.com/scylladb/scylla-operator/pkg/scheme"
+	"github.com/scylladb/scylla-operator/pkg/systemd"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/component-base/metrics/prometheus/ratelimiter"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/exec"
+)
+
+const (
+	ControllerName = "NodeSetupController"
+)
+
+var (
+	keyFunc                 = cache.DeletionHandlingMetaNamespaceKeyFunc
+	nodeConfigControllerGVK = scyllav1alpha1.GroupVersion.WithKind("NodeConfig")
+)
+
+type Controller struct {
+	kubeClient   kubernetes.Interface
+	scyllaClient scyllav1alpha1client.ScyllaV1alpha1Interface
+
+	nodeConfigLister scyllav1alpha1listers.NodeConfigLister
+
+	cachesToSync []cache.InformerSynced
+
+	eventRecorder record.EventRecorder
+
+	queue    workqueue.RateLimitingInterface
+	handlers *controllerhelpers.Handlers[*scyllav1alpha1.NodeConfig]
+
+	nodeName       string
+	nodeUID        types.UID
+	nodeConfigName string
+	nodeConfigUID  types.UID
+
+	executor           exec.Interface
+	systemdControl     *systemd.SystemdControl
+	systemdUnitManager *systemd.UnitManager
+	sysfsPath          string
+}
+
+func NewController(
+	ctx context.Context,
+	kubeClient kubernetes.Interface,
+	scyllaClient scyllav1alpha1client.ScyllaV1alpha1Interface,
+	nodeConfigInformer scyllav1alpha1informers.NodeConfigInformer,
+	nodeName string,
+	nodeUID types.UID,
+	nodeConfigName string,
+	nodeConfigUID types.UID,
+) (*Controller, error) {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartStructuredLogging(0)
+	eventBroadcaster.StartRecordingToSink(&corev1client.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
+
+	if kubeClient.CoreV1().RESTClient().GetRateLimiter() != nil {
+		err := ratelimiter.RegisterMetricAndTrackRateLimiterUsage(
+			"nodesetup_controller",
+			kubeClient.CoreV1().RESTClient().GetRateLimiter(),
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
this is not used anymore since 1.26 the metrics method is gone

Reply:
removed